prov/gni/src/gnix_rma.c

```diff
@@ -995,6 +1185,42 @@ ssize_t _gnix_rma(struct gnix_fid_ep *ep, enum gnix_fab_req_type fr_type,
 		req->flags |= GNIX_RMA_RDMA;
 	}
 
+	/* Add reads/writes to slist when FI_MORE is present, Or
+	 * if this is the first message in the chain without FI_MORE.
+	 * When FI_MORE is not present, if the slists are not empty
+	 * it is the first message without FI_MORE. */
+	if ((flags & FI_MORE) ||
+	    (!(slist_empty(&ep->more_write)) || !(slist_empty(&ep->more_read)))) {
+		req->work_fn = _gnix_rma_more_post_req;
+		if (fr_type == GNIX_FAB_RQ_RDMA_READ) {
+			slist_insert_tail(&req->rma.sle, &ep->more_read);
+		}
```

Comment:
'else' follows on the same line as a closing brace in the rest of this file.  Please match the surrounding style.